File: 프로그래머스/unrated/120924. 다음에 올 숫자/다음에 올 숫자.js
function solution(common) {
    const diff = []
    for(let i = 1; i < common.length; i++){
        diff.push(common[i] - common[i-1])
    }
    
    const vs = diff.reduce((acc, cur) => acc + cur, common[0])

    return diff[0] === diff[1] ? common[common.length-1] + diff[0] : common[common.length-1] * (diff[1] / diff[0])
}

File: 프로그래머스/unrated/120924. 다음에 올 숫자/다음에 올 숫자.py
def solution(common):
    diff = [];
    for i in range(len(common) - 1):
        diff.append(common[i+1] - common[i]);
    
    return common[len(common) - 1] + diff[0] if diff[0] == diff[1] else common[len(common) - 1] * (diff[1] / diff[0])